lib/IRremoteESP8266/src/ir_Daikin.h

```diff
@@ -128,6 +128,7 @@ const uint64_t kDaikinFirstHeader64 =
     0b1101011100000000000000001100010100000000001001111101101000010001;
 
 // Another variant of the protocol for the Daikin ARC477A1 remote.
+const uint16_t kDaikin2Freq = 36700;  // Modulation Frequency in Hz.
```

Comment:
Why are these made a global defined const variable, only to be used in a define?
This is taking more memory and it is confusing.

Reply:
For this we should ask the library maintainer  
ref: markszabo/IRremoteESP8266#635

Reply:
AFAIUI, a `const` doesn't get allocated _space_ by the compiler unless it is needed/used. Similar to a `#define`, if it gets used somewhere, it then gets referenced. Hence it's similar to a `#define` (as in, a macro) but the compiler knows what type it is and _should_ use only one reference to the value if it is used multiple times. Also, `const`s containing equations are calculated at compile-time only as they naturally can only contain constants.
See also: https://google.github.io/styleguide/cppguide.html#Use_of_const

again, AFAIUI:
```
#define FOO 0x12345
do_something(FOO);
```
and
```
const uint16_t kFoo = 0x12345;
do_something(kFoo); 
```
and
```
do_something(0x12345); 
```
i.e. Somewhere the value of 0x12345 will ultimately be stored in the compiled binary.

Given a smart enough compiler (and Gnu's ones typically are) that should produce identical code and memory usage. The difference is, for the middle one, the compiler has clear type context and can produce better diagnostics.
e.g. Generate a warning when the value is larger than the type can hold.

If I'm wrong, please let me know.

P.S. If you want me to see a reference, please use a @crankyoldgit ;-) I only just noticed this when pushing out the v2.5.6 release of the library.

Reply:
@crankyoldgit 
Well we may need to check this.
But as far as I know, these variables will be made available as an object on the system stack.
So they will use memory on the system stack (Arduino space is using 'cont' stack, the core libraries have the system stack)
It may very wel be that the compiler does see it is not referenced anywhere else, so it will then try to do some magic and not allocate it as an object on the stack, but to be sure, that has to be checked.
I don't think it will make much of a difference (if any) in the compiled binary size, but I'm more worried about the stack usage, since we've seen lots of strange errors in the last few months that appeared to be related to either one of both stacks.

The type checking is a good thing indeed.

Reply:
@TD-er I think the point is, you _shouldn't_ think of them as **variables** because they are _not_, They are **constants**, hence they are treated differently. Just like any other [literal](https://en.wikipedia.org/wiki/Literal_(computer_programming)) you'd use in the code. e,g `0x12345` in my example.

But yes. It needs to be confirmed. After all, it's the implementation that matters in the end.

Reply:
A constant is not the same as a literal.

If you use a define, then the (pre)compiler will exchange it as if it is just code duplication in the source code.
If you create an object and reference it, then there will be an object which does use resources at runtime, like some allocation on the stack. (unless it is removed due to optimizations)
For a simple int, it will probably not matter much, but for more complex objects it does matter.

In the code, the define is referencing the actual variable, so at all places where that define is being used, the code has a reference to that object.